src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx

```diff
@@ -3,223 +3,218 @@
+    });
+
+// transforms the bullets made by react-hook-forms to the objects needed for the payload to the backend
+const mapBulletsToPayload = (ls: { bulletId: number; detail: string }[]) => {
+  return ls.map((ele) => {
+    return { id: ele.bulletId, detail: ele.detail };
+  });
+};
+
+const schema = yup.object().shape({
+  name: yup.string().required('Name is required!'),
+  startDate: yup.date().required('Start Date is required!'),
+  duration: yup.number().required()
+});
 
 interface WorkPackageEditContainerProps {
   workPackage: WorkPackage;
   exitEditMode: () => void;
 }
 
-// Making this an object. Later on more functions can be used that can pass up state from inputs for wiring and such.
-export const FormContext = createContext({
-  editMode: false,
-  setField: (field: string, value: any) => {}
-});
-
 const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ workPackage, exitEditMode }) => {
   const auth = useAuth();
   const query = useQuery();
-  const { mutateAsync } = useEditWorkPackage(workPackage.wbsNum);
-  const { data: userData, isLoading, isError, error } = useAllUsers();
-
-  // states for the form's payload
-  const [projectLead, setProjectLead] = useState(workPackage.projectLead?.userId);
-  const [projectManager, setProjectManager] = useState(workPackage.projectManager?.userId);
-  const [name, setName] = useState<string>(workPackage.name);
-  const [crId, setCrId] = useState<string>(query.get('crId') || '');
-  const [startDate, setStartDate] = useState<Date>(workPackage.startDate);
-  const [duration, setDuration] = useState<number>(workPackage.duration);
-  const [deps, setDeps] = useState<WbsNumber[]>(workPackage.dependencies);
-  const [ea, setEa] = useState<{ id: number; detail: string; isResolved: boolean }[]>(
-    workPackage.expectedActivities
-      .filter((ea) => ea.dateDeleted === undefined)
-      .map((ea) => ({
-        id: ea.id,
-        detail: ea.detail,
-        isResolved: !!ea.userChecked
-      }))
-  );
-  const [dels, setDels] = useState<{ id: number; detail: string; isResolved: boolean }[]>(
-    workPackage.deliverables
-      .filter((del) => del.dateDeleted === undefined)
-      .map((d) => ({
-        id: d.id,
-        detail: d.detail,
-        isResolved: !!d.userChecked
-      }))
-  );
-  const [status, setStatus] = useState<WbsElementStatus>(workPackage.status);
-
-  const expectedActivitiesUtil: EditableTextInputListUtils = {
-    add: (val) => {
-      const clone = ea.slice();
-      clone.push({
-        id: -1,
-        detail: val,
-        isResolved: false
-      });
-      setEa(clone);
-    },
-    remove: (idx) => {
-      const clone = ea.slice();
-      clone.splice(idx, 1);
-      setEa(clone);
-    },
-    update: (idx, val) => {
-      const clone = ea.slice();
-      clone[idx].detail = val;
-      setEa(clone);
+  const allUsers = useAllUsers();
+  const { name, startDate, duration, status } = workPackage;
+  const { userId } = auth.user!;
```

Comment:
declare this later right before when you need it, and do `if (!auth.user) return LoadingIndicator ` before it so that we don't have to use `!`